tests/test-core.php

```diff
@@ -1,10 +1,85 @@
 <?php
+use Yoast\PHPUnitPolyfills\Polyfills\ExpectPHPException;
 
 // phpcs:disable PEAR.NamingConventions.ValidClassName.Invalid
+// phpcs:disable WordPress.PHP.DiscouragedPHPFunctions.runtime_configuration_error_reporting
 
 class VIP_Go__Core__Disable_Update_Caps_Test extends WP_UnitTestCase {
 	public function setUp(): void {
 		wpcom_vip_init_core_restrictions();
+		$this->error_reporting = error_reporting();
+	}
+
+	public function tearDown(): void {
+		error_reporting( $this->error_reporting );
+		parent::tearDown();
+	}
+
+	public function get_var_standard_env() {
+		define( 'VIP_ENV_VAR_MY_VAR', 'VIP_ENV_VAR_MY_VAR' );
+	}
+
+	public function get_var_legacy_env() {
+		define( 'MY_VAR', 'MY_VAR' );
+	}
+
+	// tests the use-case where $key parameter is not found
+	public function test_get_default_var() {
+		error_reporting( $this->error_reporting & ~E_USER_NOTICE );
+
+		$val = vip_get_env_var( 'MY_VAR', 'default_value' );
+		$this->assertEquals( 'default_value', $val );
+	}
+
+	/**
+	 * tests the use-case where $key parameter does not have the prefix
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test_get_var_legacy_key() {
+		error_reporting( $this->error_reporting & ~E_USER_NOTICE );
+
+		$this->get_var_legacy_env();
+		$val = vip_get_env_var( 'MY_VAR', 'default_value' );
+		$this->assertEquals( 'MY_VAR', $val );
```

Comment:
I would suggest making the value of the env var distinct from the name in test, for two reasons:

1. This is a bit confusing to read, initially I thought you were asserting that the value was equal to the name.
2. If by some future misfortune, the code accidentally returned the name instead of the value, this test would still pass.

Reply:
Thank you for the great suggestion. I have made this change in my latest commit.